File: Example/XenditExampleObjC/XenditExampleObjC/SpaceMonoFont.m
#import "SpaceMonoFont.h"
#import <CoreText/CoreText.h>

@implementation SpaceMonoFont

+ (void)register {
    NSArray<NSString *> *names = @[
        @"SpaceMono-Regular",
        @"SpaceMono-Bold",
    ];
    for (NSString *name in names) {
        NSURL *url = [[NSBundle mainBundle] URLForResource:name withExtension:@"ttf"];
        if (url) {
            CTFontManagerRegisterFontsForURL((__bridge CFURLRef)url, kCTFontManagerScopeProcess, NULL);
        } else {
            NSLog(@"[XenditExampleObjC] '%@.ttf' not found in bundle — add it to Copy Bundle Resources in Xcode.", name);
        }
    }
}

+ (UIFont *)regular { return [UIFont fontWithName:@"SpaceMono-Regular" size:14] ?: [UIFont monospacedSystemFontOfSize:14 weight:UIFontWeightRegular]; }
+ (UIFont *)bold    { return [UIFont fontWithName:@"SpaceMono-Bold"    size:14] ?: [UIFont monospacedSystemFontOfSize:14 weight:UIFontWeightBold]; }

@end

File: Example/XenditExample/XenditExample/SpaceMonoFont.swift
import CoreText
import UIKit

enum SpaceMonoFont {

    static func register() {
        let names = [
            "SpaceMono-Regular",
            "SpaceMono-Bold",
        ]
        for name in names {
            guard let url = Bundle.main.url(forResource: name, withExtension: "ttf") else {
                print("[XenditExample] '\(name).ttf' not found in bundle — add it to Copy Bundle Resources in Xcode.")
                continue
            }
            CTFontManagerRegisterFontsForURL(url as CFURL, .process, nil)
        }
    }

    static var regular: UIFont { UIFont(name: "SpaceMono-Regular", size: 14) ?? .monospacedSystemFont(ofSize: 14, weight: .regular) }
    static var bold:    UIFont { UIFont(name: "SpaceMono-Bold",    size: 14) ?? .monospacedSystemFont(ofSize: 14, weight: .bold) }
}
